File: app/src/main/java/com/musicplayer/MusicApplication.java
package com.musicplayer;

import android.app.Application;
import android.util.Log;
import com.musicplayer.downloader.OkHttpDownloader;
import com.yausername.youtubedl_android.YoutubeDL;
import org.schabi.newpipe.extractor.NewPipe;

public class MusicApplication extends Application {
    private static final String TAG = "DEBUG_PLAYER";

    @Override
    public void onCreate() {
        super.onCreate();
        
        try {
            YoutubeDL.getInstance().init(this);
            // Actualizar a la última versión de yt-dlp cada vez que inicia la app para evitar bloqueos de YouTube
            new Thread(() -> {
                try {
                    Log.d(TAG, "Application: Buscando actualizaciones de yt-dlp...");
                    YoutubeDL.getInstance().updateYoutubeDL(this, YoutubeDL.UpdateChannel._STABLE);
                    Log.d(TAG, "Application: yt-dlp actualizado con éxito.");
                } catch (Exception e) {
                    Log.e(TAG, "Application: Error actualizando yt-dlp", e);
                }
            }).start();

            // Inicializar NewPipe para búsquedas ultra-rápidas
            NewPipe.init(new OkHttpDownloader());
            Log.d(TAG, "Application: Motores listos.");
        } catch (Exception e) {
            Log.e(TAG, "Application: Error inicializando motores", e);
        }
    }
}

File: app/src/main/java/com/musicplayer/MusicApplication.kt
package com.musicplayer

import android.app.Application
import android.util.Log
import com.musicplayer.downloader.OkHttpDownloader
import com.musicplayer.utils.CrashHandler
import com.yausername.ffmpeg.FFmpeg
import com.yausername.youtubedl_android.YoutubeDL
import org.schabi.newpipe.extractor.NewPipe
import java.util.concurrent.Executors

class MusicApplication : Application() {
    companion object {
        private const val TAG = "DEBUG_PLAYER"
    }

    override fun onCreate() {
        super.onCreate()
        CrashHandler.init(this)
        
        try {
            Log.d(TAG, "Application: Inicializando YoutubeDL...")
            YoutubeDL.getInstance().init(this)
            Log.d(TAG, "Application: Inicializando FFmpeg...")
            FFmpeg.getInstance().init(this)
            
            // Actualizar a la última versión de yt-dlp cada vez que inicia la app para evitar bloqueos de YouTube
            Executors.newSingleThreadExecutor().execute {
                try {
                    Log.d(TAG, "Application: Buscando actualizaciones de yt-dlp...")
                    YoutubeDL.getInstance().updateYoutubeDL(this, YoutubeDL.UpdateChannel._NIGHTLY)
                    Log.d(TAG, "Application: yt-dlp actualizado con éxito.")
                } catch (e: Exception) {
                    Log.e(TAG, "Application: Error (no crítico) actualizando yt-dlp", e)
                }
            }

            // Inicializar NewPipe para búsquedas ultra-rápidas
            Log.d(TAG, "Application: Inicializando NewPipe...")
            NewPipe.init(OkHttpDownloader())
            Log.d(TAG, "Application: Motores listos ✅")
        } catch (e: Exception) {
            Log.e(TAG, "Application: ERROR FATAL EN INICIALIZACIÓN", e)
        }
    }
}
